test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -348,6 +348,14 @@ protected final boolean enableWarningsCheck() {
         return false;
     }
 
+    @Override
+    protected boolean enableSearchContextAfterCheck() {
+        // We don't want to check the release of search contexts between tests if the cluster is shared.
```

Comment:
I think debugging would be easier if we return true here and false in AsyncSearchActionIT instead.

Reply:
I reverted this since this solution doesn't work. Sorry for the trouble